cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceConfigMapUtilsTest.java

```diff
@@ -0,0 +1,173 @@
+                BROKER_LOAD_KEY, BROKER_LOAD));
+        updateRebalanceConfigMapWithProgressFields(KafkaRebalanceState.Stopped, es, cm);
+
+        m =  cm.getData();
+        assertThat(m.containsKey(ESTIMATED_TIME_TO_COMPLETION_IN_MINUTES_KEY), is(false));
+        assertThat(m.containsKey(COMPLETED_BYTE_MOVEMENT_PERCENTAGE_KEY), is(false));
+        assertThat(m.containsKey(EXECUTOR_STATE_KEY), is(false));
+        assertThat(m.get(BROKER_LOAD_KEY), is(BROKER_LOAD));
+    }
+
+    @Test
+    public void testProgressFieldsForProposalNotReady() throws Exception {
+        ExecutorStatus es = new ExecutorStatus(createExecutorStatusJson(
+                CruiseControlExecutorState.NO_TASK_IN_PROGRESS.toString(), "250", "10000", null));
+        ConfigMap cm = createKafkaRebalanceConfigMap(Map.of(ESTIMATED_TIME_TO_COMPLETION_IN_MINUTES_KEY, "5",
+                COMPLETED_BYTE_MOVEMENT_PERCENTAGE_KEY, "100",
+                EXECUTOR_STATE_KEY, es.toString(),
+                BROKER_LOAD_KEY, BROKER_LOAD));
+        updateRebalanceConfigMapWithProgressFields(KafkaRebalanceState.NotReady,  es, cm);
+
+        Map<String, String> m =  cm.getData();
+        assertThat(m.containsKey(ESTIMATED_TIME_TO_COMPLETION_IN_MINUTES_KEY), is(false));
+        assertThat(m.get(COMPLETED_BYTE_MOVEMENT_PERCENTAGE_KEY), is("100"));
+        assertThat(m.get(EXECUTOR_STATE_KEY), is(es.toString()));
+        assertThat(m.get(BROKER_LOAD_KEY), is(BROKER_LOAD));
+
+        cm = createKafkaRebalanceConfigMap(Map.of(ESTIMATED_TIME_TO_COMPLETION_IN_MINUTES_KEY, "5",
+                BROKER_LOAD_KEY, BROKER_LOAD));
+        updateRebalanceConfigMapWithProgressFields(KafkaRebalanceState.NotReady, es, cm);
+
+        m =  cm.getData();
+        assertThat(m.containsKey(ESTIMATED_TIME_TO_COMPLETION_IN_MINUTES_KEY), is(false));
+        assertThat(m.containsKey(COMPLETED_BYTE_MOVEMENT_PERCENTAGE_KEY), is(false));
+        assertThat(m.containsKey(EXECUTOR_STATE_KEY), is(false));
+        assertThat(m.get(BROKER_LOAD_KEY), is(BROKER_LOAD));
+    }
+
+    @Test
+    public void testProgressFieldsForReadyState() throws Exception {
+        ExecutorStatus es = new ExecutorStatus(createExecutorStatusJson(
+                CruiseControlExecutorState.NO_TASK_IN_PROGRESS.toString(), null, null, null));
+        ConfigMap cm = createKafkaRebalanceConfigMap(BROKER_LOAD_MAP);
+        updateRebalanceConfigMapWithProgressFields(KafkaRebalanceState.Ready, es, cm);
+
+        Map<String, String> m =  cm.getData();
+        assertThat(m.get(ESTIMATED_TIME_TO_COMPLETION_IN_MINUTES_KEY), is(TIME_COMPLETED));
+        assertThat(m.get(COMPLETED_BYTE_MOVEMENT_PERCENTAGE_KEY), is(BYTE_MOVEMENT_COMPLETED));
+        assertThat(m.containsKey(EXECUTOR_STATE_KEY), is(false));
+        assertThat(m.get(BROKER_LOAD_KEY), is(BROKER_LOAD));
+    }
+
+    @Test
+    public void testProgressFieldsForUnsupportedStates() throws Exception {
```

Comment:
nit: I think `unsupportedStates` is a bit confusing. Maybe better to have consistent names so something like `testProgressFieldsForNewAndPendingProposalStates`?

Reply:
Fixed.